### PR TITLE
DROOLS-6213: Removed Critical Bugs in drools-wb

### PR DESCRIPTION
**Pull request by yesamer**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6213

Fixing SONAR reported CRITICAL BUGS https://sonarcloud.io/project/issues?id=org.drools%3Adrools-wb&resolved=false&severities=CRITICAL&types=BUG

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by jomarko** [comment]
Jenkins do fdb

**Review by Rikkola — APPROVED** [review]
Looks cleaner and passes tests. All good.

File: drools-wb-screens/drools-wb-scorecard-xls-editor/drools-wb-scorecard-xls-editor-backend/src/main/java/org/drools/workbench/screens/scorecardxls/backend/server/ScoreCardXLSServiceImpl.java [hunk 1]
```diff
@@ -146,22 +137,17 @@ private Path writeToFile( final Path resource,
...
+            ioService.createFile( nioPath );
         }
-        log.info( "USER:" + sessionInfo.getIdentity().getIdentifier() + " " + userAction + " asset [" + resource.getFileName() + "]" );
-
-        OutputStream outputStream = null;
-        try {
-            final org.uberfire.java.nio.file.Path nioPath = Paths.convert( resource );
-            if ( create ) {
-                ioService.createFile( nioPath );
-            }
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-scorecard-xls-editor/drools-wb-scorecard-xls-editor-backend/src/main/java/org/drools/workbench/screens/scorecardxls/backend/server/ScoreCardXLSServiceImpl.java:160]: Should we move both:
- `Paths.convert(resource)`
- `ioService.createFile(nioPath)`

out of the try catch block? Especially the second one informs in method signature about quite a lot possible runtime exceptions.

What is your opinion? Was this change intentional? I know general rule is to keep try catch block scope as small as possible. However here we update existing code base what makes our situation difficult. Did you compared what happens when user tries to upload a xls scorecard that has an existing name before and after your PR please?

**Review by jomarko — COMMENTED** [review]
@yesamer thank you for this clean-up.

### Code review
One question inline

### Manual review
- Tried to convert GDST to XLS decision table
- Tried to open XLS decision table
- Tried to validate XLS decision table

No issues found.

File: drools-wb-screens/drools-wb-scorecard-xls-editor/drools-wb-scorecard-xls-editor-backend/src/main/java/org/drools/workbench/screens/scorecardxls/backend/server/ScoreCardXLSServiceImpl.java (hunk 1 above)
Comment (yesamer) [reply drools-wb-screens/drools-wb-scorecard-xls-editor/drools-wb-scorecard-xls-editor-backend/src/main/java/org/drools/workbench/screens/scorecardxls/backend/server/ScoreCardXLSServiceImpl.java:160]: @jomarko  The change was intentional, and the reason for that change is `ioService.newOutputStream()` methods requires `nioPath` as paramenter. 
Please consider `ExceptionUtilities.handleException( e );`, which is the action performed if an expection is catched.
If the expection is unchecked (instance of RuntimeExpection), it is simply re-thrown, and this is exactly the case of the two methods you mentioned.

Comment (yesamer) [reply drools-wb-screens/drools-wb-scorecard-xls-editor/drools-wb-scorecard-xls-editor-backend/src/main/java/org/drools/workbench/screens/scorecardxls/backend/server/ScoreCardXLSServiceImpl.java:160]: The difference I see is the logger message inside `ExceptionUtilities.handleException( e );` will be no more reached. 
So, I don't see problems to move 
       ` if ( create ) {
            ioService.createFile( nioPath );
        }`
Inside the `try-catch` for this resaon.
`nioPath` I don't see a way to move it in `try-catch`.

Comment (jomarko) [reply drools-wb-screens/drools-wb-scorecard-xls-editor/drools-wb-scorecard-xls-editor-backend/src/main/java/org/drools/workbench/screens/scorecardxls/backend/server/ScoreCardXLSServiceImpl.java:160]: ok, thank you @yesamer

**Review by jomarko — APPROVED** [review]

**Review by karreiro — APPROVED** [review]

**Pull request merged**